protoc_plugin/lib/src/message_generator.dart

```diff
@@ -430,6 +431,42 @@ class MessageGenerator extends ProtobufContainer {
     out.println();
   }
 
+  void _generateFactory(IndentingWriter out) {
+    if (!fileGen.options.disableConstructorArgs && _fieldList.isNotEmpty) {
+      out.println('factory $classname({');
+      for (final field in _fieldList) {
+        _emitDeprecatedIf(field.isDeprecated, out);
+        if (field.isRepeated && !field.isMapField) {
+          out.println(
+              '  ${field.baseType.getRepeatedDartTypeIterable(fileGen)}? ${field.memberNames!.fieldName},');
+        } else {
+          out.println(
+              '  ${field.getDartType()}? ${field.memberNames!.fieldName},');
+        }
+      }
+      out.println('}) {');
+      out.println('  final result = create();');
```

Comment:
Note that this local variable (`result`) can conflict with named params corresponding to a proto field.

For dartpad's protos code, we're seeing:

```
A value of type 'CompileResponse' can't be assigned to a variable of type 'String'.
Try changing the type of the variable, or casting the right-hand type to 'String'.
```

Perhaps name this `_result`? `$result`?

Reply:
Our proto definition:

```
message CompileResponse {
  string result = 1;
  string sourceMap = 2;

  // Make this response compatible with BadRequest
  ErrorMessage error = 99;
}
```

Reply:
https://github.com/google/protobuf.dart/pull/869